rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -716,12 +746,26 @@ rmw_ret_t rmw_destroy_node(rmw_node_t * node)
 
   Participant * participant = impl->participant;
 
+  // Begin deleting things in the same order they were created in rmw_create_node().
   std::pair<StatefulReader *, StatefulReader *> EDPReaders = participant->getEDPReaders();
-  EDPReaders.first->setListener(nullptr);
-  delete (impl->secondarySubListener);
-  EDPReaders.second->setListener(nullptr);
-  delete (impl->secondaryPubListener);
+  if (!EDPReaders.first->setListener(nullptr)) {
+    RMW_SET_ERROR_MSG("failed to unset EDPReader listener");
+    return RMW_RET_ERROR;
+  }
+  delete impl->secondarySubListener;
+  if (!EDPReaders.second->setListener(nullptr)) {
+    RMW_SET_ERROR_MSG("failed to unset EDPReader listener");
+    return RMW_RET_ERROR;
+  }
+  delete impl->secondaryPubListener;
 
+  if (node) {
+    free(const_cast<char *>(node->name));
+    free(const_cast<char *>(node->namespace_));
```

Comment:
We do this in existing code for debugging purposed. If the caller passes a pointer and you watch it in the debugger you will see that the member are actually reset to a `nullptr` rather than still contain a memory address.